src/Infrastructure/Dashboard/TermMeta/TermMetaRegistrar.php

```diff
@@ -64,21 +72,83 @@ function () use ($termMetaFields): void {
 
         add_action("edit_{$taxonomy}", $updateCallback);
         add_action("create_{$taxonomy}", $updateCallback);
+        add_filter('pre_insert_term', $this->buildValidationCallback(...$termMetaFields));
+    }
+
+    // phpcs:ignore Inpsyde.CodeQuality.NestingLevel.High
+    private function buildValidationCallback(TermMetaField ...$termMetaFields): callable
```

Comment:
I think I would move this and the `checkErrors()` methods to their own class to better separate concerns. However based on the fact that the `updateCallback` also lives here, feel free to ignore this suggestion.